takeshape/takeshape-starter-ecommerce-kitchen-sink
# add product reviews from reviews io to sample

The reviews.io api bits, and some general build chores (bump deps, some TS conversions).

The `webhook` role is too open, but we can tighten it up later.

### Test Plan (Steps to test):

1. Check out the deployed site
1. Review the code
1. Try these queries in the shared TakeShape project

```
{
  getProductReviews(sku: "prod_LNjQn6SmhBfhkT") {
    stats {
      average
      count
    }
		reviews {
      data {
        title
        rating
        reviewer {
          first_name
          last_name
        }
      }
    }
  }
}
``` 

```
mutation {
  queueReviewInvitation(
    name:"RANDOM", 
    email: "YOURNAME+test@takeshape.io",
  	orderId: "RANDOM",
    products: [{
      name: "Pink Camo Socks",
      sku: "prod_LNjQn6SmhBfhkT"
      description: "Are these pink camo socks perfect for a night out on the town, or what?",
      pageUrl: "https://ecommerce-kitchen-sink.vercel.app/product/prod_LNjQn6SmhBfhkT"
    }]
  ) {
    status
    messages
  }
}
```

### Checklist:
